packages/fuselage-hooks/src/useStorage.ts

```diff
@@ -24,49 +37,56 @@ function makeStorageHook(
     const initialValueRef = useRef(initialValue);
     initialValueRef.current = initialValue;
 
-    const [storedValue, setStoredValue] = useState<T>(() => {
-      if (!storage) {
-        return initialValue;
-      }
+    const valueRef = useRef(getStorageItem(getKey(key), initialValue, storage));
```

Comment:
During re-renders (caused by external reasons), `getStorageItem` will be called again.